### PR TITLE
[AIRLFOW-6024] Do not use the logger in CLI

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6024
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by turbaszek — COMMENTED** [review]
@mik-laj why do we want to use printing?

**Comment by mik-laj** [comment]
There is no need to use the logger because all messages are intended to be displayed on the console. This should not be saved to a file or other external system e.g. Stackdriver. If you need to write to a file, you can redirect the stream.

We want the messages also to be displayed also when the following command is used.
```
AIRFLOW__CORE__LOG_LEVEL=fatal airflow pools list
```

File: airflow/cli/commands/dag_command.py [hunk 1]
```diff
@@ -104,16 +104,15 @@ def dag_trigger(args):
...
     try:
         message = api_client.trigger_dag(dag_id=args.dag_id,
                                          run_id=args.run_id,
                                          conf=args.conf,
                                          execution_date=args.exec_date)
+        print(message)
     except OSError as err:
-        log.error(err)
+        print(str(err))
         raise AirflowException(err)
```
Comment (turbaszek) [inline airflow/cli/commands/dag_command.py:114]: Do we need print here? The same exception with message will be rise.

File: airflow/cli/commands/dag_command.py [hunk 2]
```diff
@@ -125,16 +124,15 @@ def dag_delete(args):
...
     if args.yes or input(
             "This will drop all existing records related to the specified DAG. "
             "Proceed? (y/n)").upper() == "Y":
         try:
             message = api_client.delete_dag(dag_id=args.dag_id)
+            print(message)
         except OSError as err:
-            log.error(err)
+            print(str(err))
             raise AirflowException(err)
```
Comment (turbaszek) [inline airflow/cli/commands/dag_command.py:133]: Do we need print here? The same exception with message will be rise.

**Review by kaxil — APPROVED** [review]

File: airflow/cli/commands/dag_command.py (hunk 1 above)
Comment (mik-laj) [reply airflow/cli/commands/dag_command.py:114]: Fixed. Raising exceptions is also not the best solution, but it is another problem.

File: airflow/cli/commands/dag_command.py (hunk 2 above)
Comment (mik-laj) [reply airflow/cli/commands/dag_command.py:133]: Fixed. Raising exceptions is also not the best solution, but it is another problem.

**Pull request merged**